manifests/quadlet_file.pp

```diff
@@ -0,0 +1,113 @@
+# @summary Creates a systemd Podman Quadlet file
+# Quadlet will generate a unit file, and this service can be managed by puppet.
+# @api public
+#
+# @see podman.systemd.unit(5)
+#
+# @param name
+#   The name of the quadlet file
+#
+# @param ensure
+#   The state of the quadlet file to ensure
+#
+# @param content
+#   The full content of the quadlet file
+#
+# @param path
+#   The path where the quadlet file will be created
+#   For systemd in user mode use any of
+#   - ~/.config/containers/systemd
+#   - /etc/containers/systemd/users/$(UID)
+#
+#   For global systemd use any of:
+#   - /etc/containers/systemd
+#   - /usr/share/containers/systemd
+#
+# @param source
+#   The ``File`` resource compatible ``source``
+#
+#   * Mutually exclusive with ``$content``
+#
+# @param owner
+#   The owner to set on the unit file
+#
+# @param group
+#   The group to set on the unit file
+#
+# @param mode
+#   The mode to set on the unit file
+#
+# @param enable
+#   If set, will manage the unit enablement status.
+#
+# @param active
+#   If set, will manage the state of the unit.
+#
+# @param restart
+#   Specify a restart command manually. If left unspecified, a standard Puppet service restart happens.
+#
+# @param service_parameters
+#   hash that will be passed with the splat operator to the service resource
+#
+# @param daemon_reload
+#   call `systemd::daemon-reload` to ensure that the modified unit file is loaded
+#
+# @param service_restart
+#   restart (notify) the service when unit file changed
+define systemd::quadlet_file (
+  Enum['present', 'absent']                $ensure    = 'present',
+  Stdlib::Absolutepath                     $path      = '/etc/containers/systemd',
+  Optional[Variant[String, Sensitive[String], Deferred]] $content = undef,
+  Optional[String]                         $source    = undef,
+  String                                   $owner     = 'root',
+  String                                   $group     = 'root',
```

Comment:
can you please update those so we don't allow empty string? `String[1]`

Reply:
Done.
Note: it was copy-pasted from `unit_file.pp`